packages/graphql-lowcode/src/generate/generateGraphqlQueries.ts

```diff
@@ -22,79 +24,212 @@ interface IntrospectionQuery {
+
+      parameters = [...parameters, newParameter]
+    })
+  }
+
+  return parameters.length ? `(${parameters.join(', ')})` : ''
+}
+
+function buildSelectQuery(queryRoot: Root): string {
+  let selectQueries: string[] = []
+
+  queryRoot.fields.forEach(field => {
+    const fragmentName = `${field.name}_${field.type.name}`
+    const parametersString = buildParametersString(field.args, queryRoot.kind)
+    const pkParameter = queryRoot.kind !== 'LIST' ? '($id: ID!)' : ''
+
+    const newSelectQuery = `query ${field.name}${pkParameter} {\n  ${field.name}${parametersString} {\n    ...${fragmentName}\n  }\n}`
+
+    selectQueries = [...selectQueries, newSelectQuery]
+  })
+
+  return selectQueries.join('\n\n')
+}
+
+function buildMutationQuery(mutationRoot: Root): string {
+  let mutationQueries: string[] = []
+
+  mutationRoot.fields.forEach(field => {
+    const mutationType = getMutationType(field.args)
+
+    const fragmentName = `${field.name}_${field.type.name}`
 
-  fields.forEach(field => {
-    //skip non scalar fields
-    if (field.type && field.name && field.type.kind === 'SCALAR') scalarFields = [...scalarFields, field.name]
+    mutationQueries = [...mutationQueries, buildMutationString(field, fragmentName)]
   })
 
-  return scalarFields
+  return mutationQueries.join('\n\n')
+
 }
 
-function buildSelectQuery(queryRoot: QueryRoot) {
-  const fragmentName = `${queryRoot.queryName}_${queryRoot.entityName}`
+function buildMutationString(field: Field, fragmentName: string) {
+  const parametersQuery = buildParametersString(field.args, '')
 
-  return `query ${queryRoot.queryName} {\n  ${queryRoot.queryName}(limit: 100) {\n    ...${fragmentName}\n  }\n}`
+  return `mutation ${field.name} {\n  ${field.name}${parametersQuery} {\n    ...${fragmentName}\n  }\n}`
 }
 
-function buildFragmentQuery(entityFields: Field[], queryRoot: QueryRoot) {
-  const scalarFields = getScalarFields(entityFields)
+function buildReturningString(types: TypesObject[], returningType: string): string {
+  let returningFields: string[] = []
+
+  for(const type of types) {
+    if(type.name === returningType) {
+      type.fields.forEach(field => {
+        returningFields = [...returningFields, field.name]
+      })
 
-  return `fragment ${queryRoot.queryName}_${queryRoot.entityName} on ${queryRoot.entityName} {\n  ${scalarFields.join('\n  ')}\n}`
+      break
+    }
+  }
+
+  return `returning {\n    ${returningFields.join('\n    ')}\n  }`
 }
 
-function buildGraphqlQuery(selectQuery: string, fragmentQuery: string) {
-  return `${selectQuery}\n\n${fragmentQuery}`
+function buildFragmentsQuery(entities: Entity[], types: TypesObject[]): string {
+  let fragmentsStrings: string[] = []
+
+  entities.forEach(entity => {
+    let fragmentFields: string[] = []
+
+    entity.fields.forEach(field => {
+      //if mutation root has returning type, build returning object with fields to be inserted into fragment fields
+      if (field.name === 'returning') {
```

Comment:
`'returning'` is Hasura specific...
- either: paramterize your generic algorithm with hasura specific logic
- or just try to loop inside